velox/connectors/hive/SplitReader.cpp

```diff
@@ -169,28 +245,35 @@ void SplitReader::prepareSplit(
 
   baseReader_ = dwio::common::getReaderFactory(baseReaderOpts_.getFileFormat())
                     ->createReader(std::move(baseFileInput), baseReaderOpts_);
+}
 
-  // Note that this doesn't apply to Hudi tables.
+bool SplitReader::testEmptySplit(
+    dwio::common::RuntimeStatistics& runtimeStats) {
   emptySplit_ = false;
```

Comment:
createReader might have set emptySplit_ to true and do we want to revert? And do we support to reuse the same SplitReader on the same split? If not, let's have more sanity check here? Thanks!

Reply:
@xiaoxmeng Thanks for catching this. One SplitReader object is created for each split, and `prepareSplit()` is supposed to be called once only. I have made the following change here:
```
bool SplitReader::splitIsEmpty(dwio::common::RuntimeStatistics& runtimeStats) {
  // emptySplit_ may already be set if the data file is not found. In this case
  // we don't need to test further.
  if (emptySplit_) {
    return true;
  }
...
```